### PR TITLE
Add CORS Valve fixes

### PR DESCRIPTION
**Pull request by ivantha**
### Proposed changes in this pull request
$subject.

Fixes https://github.com/wso2/product-is/issues/8551

### When to merge
After https://github.com/wso2/carbon-identity-framework/pull/3029

### CODE REVIEW

File: components/org.wso2.carbon.identity.cors.service/src/main/java/org/wso2/carbon/identity/cors/service/internal/impl/CORSManagerImpl.java [hunk 1]
```diff
@@ -121,6 +137,34 @@ public CORSConfiguration getCORSConfiguration(String tenantDomain) throws CORSMa
...
 
+    /**
+     * Select highest priority CORSOrigin DAO from an already sorted list of CORSOrigin DAOs.
+     *
+     * @return Highest priority CORSOrigin DAO.
+     */
+    private CORSOriginDAO getCORSOriginDAO() throws CORSManagementServiceException {
+
+        if (!this.corsOriginDAOS.isEmpty()) {
+            return corsOriginDAOS.get(corsOriginDAOS.size() - 1);
```
Comment (ashensw) [inline components/org.wso2.carbon.identity.cors.service/src/main/java/org/wso2/carbon/identity/cors/service/internal/impl/CORSManagerImpl.java:148]: So the highest number denoting the higher priority? Is the `corsOriginDAOS` list a sorted list based on the priority? Can you add a comment explaining how the priority order get resolved?

File: components/org.wso2.carbon.identity.cors.service/src/main/java/org/wso2/carbon/identity/cors/service/internal/impl/CORSManagerImpl.java [hunk 2]
```diff
@@ -121,6 +137,34 @@ public CORSConfiguration getCORSConfiguration(String tenantDomain) throws CORSMa
...
+
+    /**
+     * Select highest priority CORSConfiguration DAO from an already sorted list of CORSConfiguration DAOs.
+     *
+     * @return Highest priority CORSConfiguration DAO.
+     */
+    private CORSConfigurationDAO getCORSConfigurationDAO() throws CORSManagementServiceException {
+
+        if (!this.corsConfigurationDAOS.isEmpty()) {
+            return corsConfigurationDAOS.get(corsConfigurationDAOS.size() - 1);
```
Comment (ashensw) [inline components/org.wso2.carbon.identity.cors.service/src/main/java/org/wso2/carbon/identity/cors/service/internal/impl/CORSManagerImpl.java:162]: Same as the above comment.

File: components/org.wso2.carbon.identity.cors.service/pom.xml [hunk 3]
```diff
@@ -68,26 +68,21 @@
...
+                            org.wso2.carbon.identity.configuration.mgt.core.exception;version="${carbon.identity.package.import.version.range}",
+                            org.wso2.carbon.identity.configuration.mgt.core.model;version="${carbon.identity.package.import.version.range}",
+                            org.wso2.carbon.identity.application.authentication.framework.util;version="${carbon.identity.package.import.version.range}",
+                            org.wso2.carbon.identity.application.common;version="${carbon.identity.package.import.version.range}",
                             org.wso2.carbon.identity.application.common.cache;version="${carbon.identity.package.import.version.range}",
                             org.osgi.framework; version="${osgi.framework.imp.pkg.version.range}",
                             org.osgi.service.component; version="${osgi.service.component.imp.pkg.version.range}",
-                            org.apache.commons.lang; version="${commons-lang.wso2.version}",
+                            <!--org.apache.commons.lang; version="${commons-lang.wso2.version}",-->
```
Comment (ashensw) [inline components/org.wso2.carbon.identity.cors.service/pom.xml:85]: Let's remove commented lines if not applicable.

File: components/org.wso2.carbon.identity.cors.service/src/main/java/org/wso2/carbon/identity/cors/service/internal/CORSServiceHolder.java [hunk 4]
```diff
@@ -19,17 +19,21 @@
...
+import java.util.ArrayList;
+import java.util.List;
 
 /**
  * Service holder class for CORS-Service.
  */
 public class CORSServiceHolder {
 
+    private List<CORSOriginDAO> corsOriginDAOS = new ArrayList<>();
+    private List<CORSConfigurationDAO> corsConfigurationDAOS = new ArrayList<>();
```
Comment (malithie) [inline components/org.wso2.carbon.identity.cors.service/src/main/java/org/wso2/carbon/identity/cors/service/internal/CORSServiceHolder.java:34]: Why we managed to add multiple configuration DAOs?

File: components/org.wso2.carbon.identity.cors.service/src/main/java/org/wso2/carbon/identity/cors/service/internal/impl/CORSManagerImpl.java (hunk 1 above)
Comment (ivantha) [reply components/org.wso2.carbon.identity.cors.service/src/main/java/org/wso2/carbon/identity/cors/service/internal/impl/CORSManagerImpl.java:148]: Removed this feature.

File: components/org.wso2.carbon.identity.cors.service/src/main/java/org/wso2/carbon/identity/cors/service/internal/CORSServiceHolder.java (hunk 4 above)
Comment (ivantha) [reply components/org.wso2.carbon.identity.cors.service/src/main/java/org/wso2/carbon/identity/cors/service/internal/CORSServiceHolder.java:34]: Removed this feature.

Comment (ivantha) [reply components/org.wso2.carbon.identity.cors.service/src/main/java/org/wso2/carbon/identity/cors/service/internal/CORSServiceHolder.java:34]: Removed this feature.

File: components/org.wso2.carbon.identity.cors.service/src/main/java/org/wso2/carbon/identity/cors/service/internal/store/ServerCORSStore.java [hunk 5]
```diff
@@ -46,6 +46,7 @@
         if (serverCORSOrigins == null) {
             List<ValidatedOrigin> validatedOrigins = new ArrayList<>();
 
+            // TODO : fix
```
Comment (ashensw) [inline components/org.wso2.carbon.identity.cors.service/src/main/java/org/wso2/carbon/identity/cors/service/internal/store/ServerCORSStore.java:49]: Do we have any todo?

**Pull request merged**